compiler/rustc_resolve/src/imports.rs

```diff
@@ -212,25 +210,32 @@ pub(crate) struct NameResolution<'a> {
     /// Single imports that may define the name in the namespace.
     /// Imports are arena-allocated, so it's ok to use pointers as keys.
     pub single_imports: FxHashSet<Interned<'a, Import<'a>>>,
-    /// The least shadowable known binding for this name, or None if there are no known bindings.
-    pub binding: Option<&'a NameBinding<'a>>,
-    pub shadowed_glob: Option<&'a NameBinding<'a>>,
+    non_glob_binding: Option<&'a NameBinding<'a>>,
+    glob_binding: Option<&'a NameBinding<'a>>,
 }
 
 impl<'a> NameResolution<'a> {
-    /// Returns the binding for the name if it is known or None if it not known.
     pub(crate) fn binding(&self) -> Option<&'a NameBinding<'a>> {
-        self.binding.and_then(|binding| {
-            if !binding.is_glob_import() || self.single_imports.is_empty() {
-                Some(binding)
-            } else {
-                None
-            }
+        self.non_glob_binding()
+            .or_else(|| if self.single_imports.is_empty() { self.glob_binding() } else { None })
```

Comment:
This logic is likely incorrect and is a source of issues like https://github.com/rust-lang/rust/issues/56593 (because it doesn't account for macros), but let's keep it as is in this PR.

I'd just want to rename it to something more explanatory, like `determinate_binding`, keep the removed comment and add a FIXME.

Reply:
Yes, I think that issue #56593 is interesting and could be the next task to work on.